tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/annotations/annotation.bal

```diff
@@ -27,6 +27,64 @@ function testAnnotOnBoundMethod() {
     assertEquality(102, rec.i);
 }
 
+type AnnotationRecord record {|
+    int minValue;
+    int maxValue;
+|};
+
+annotation AnnotationRecord BarAnnotation on type;
+annotation AnnotationRecord BarAnnotation2 on type;
+
+@BarAnnotation {
+    minValue: 18,
+    maxValue: 36
+}
+type Bar int|float;
+type Bar2 int|float;
+
+@BarAnnotation2 {
+    minValue: 18,
+    maxValue: 36
+}
+type Bar3 Bar4;
+type Bar4 int|float;
+
+function testAnnotationsOnTypeRefTypes() {
+    var f1 = function (typedesc td) {
+        AnnotationRecord? 'annotation = td.@BarAnnotation;
+        assertTrue('annotation is AnnotationRecord);
+        AnnotationRecord rec = <AnnotationRecord> 'annotation;
+        assertEquality(rec.minValue, 18);
+        assertEquality(rec.maxValue, 36);
```

Comment:
I would have the anonymous function return the annotation and do the assertions outside. Right now, it's like we are calling a test from within a test.

Reply:
updated